core/src/main/java/com/google/instrumentation/trace/TraceConfig.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Global configuration of the trace service. This allows users to change configs for the default
+ * sampler, maximum events to be kept, etc (see {@link TraceParams} for details).
+ */
+public abstract class TraceConfig {
+  private static final NoopTraceConfig noopTraceConfig = new NoopTraceConfig();
+
+  /**
+   * Returns the active {@code TraceParams}.
+   *
+   * @return the active {@code TraceParams}.
+   */
+  public abstract TraceParams getActiveTraceParams();
+
+  /**
+   * Updates the active {@link TraceParams} permanently.
+   *
+   * @param traceParams the new active {@code TraceParams}.
+   */
+  public abstract void permanentlyUpdateActiveTraceParams(TraceParams traceParams);
```

Comment:
Permanently makes it sound like the TraceParams cannot be set again.  Would it be better to name the methods updateActiveTraceParams and temporarilyUpdateActiveTraceParams?